src/mono/mono/metadata/threads.c

```diff
@@ -4895,12 +4896,12 @@ alloc_context_static_data_helper (gpointer key, gpointer value, gpointer user)
 }
 
 static StaticDataFreeList*
-search_slot_in_freelist (StaticDataInfo *static_data, guint32 size, guint32 align)
+search_slot_in_freelist (StaticDataInfo *static_data, guint32 size, gint32 align)
 {
 	StaticDataFreeList* prev = NULL;
 	StaticDataFreeList* tmp = static_data->freelist;
 	while (tmp) {
-		if (tmp->size == size) {
+		if (tmp->size == size && tmp->align == align) {
```

Comment:
this could be `tmp->align >= align`, right?

Reply:
I was debating if that would be okay.

Reply:
Although, I guess the same could be said for size.

Reply:
With `>=` it would be closer to what we do today (in that slots are more likely to be reused).

I think either way is fine unless we see evidence of pathological behavior.